packages/typespec-vscode/src/vscode-cmd/emit-code/emit-code.ts

```diff
@@ -854,3 +858,102 @@ async function isCompilerSupport(client: TspLanguageClient): Promise<boolean> {
   }
   return true;
 }
+
+function getConfigEntriesFromEmitterOptions(
+  emitterOptions: Record<string, any>,
+): Record<string, any> {
+  const configEntries: Record<string, any> = {};
+  if (!emitterOptions.properties) {
+    return configEntries;
+  }
+
+  for (const [propertyName, propertySchema] of Object.entries(emitterOptions.properties)) {
+    const {
+      description = "",
+      type,
+      enum: enumValues,
+      default: defaultValue,
+    } = propertySchema as any;
+
+    const commentParts: string[] = [];
+    if (description) commentParts.push(description);
+    if (type) commentParts.push(`Type: ${type}`);
+    if (Array.isArray(enumValues)) commentParts.push(`Options: ${enumValues.join(", ")}`);
+    const comment = commentParts.join(" ");
+
+    let value = defaultValue;
+    if (value === undefined) {
+      // If there is no default value; provide default values according to the following implementations based on the type.
+      switch (type) {
+        case "string":
+          value = `""`;
+          break;
+        case "int":
+        case "number":
+          value = 0;
+          break;
+        case "object":
+          value = `{}`;
+          break;
+        case "bool":
+        case "boolean":
+          value = false;
+          break;
+        case "array":
+          value = [];
+          break;
+      }
+    }
+
+    configEntries[propertyName] = { value, comment };
+  }
+
+  return configEntries;
+}
+
+async function generateAnnotatedYamlFile(
+  configYaml: Document,
+  packageName: string,
+  baseDir: string,
+): Promise<void> {
+  try {
+    const emitterOptions = await loadEmitterOptions(baseDir, packageName);
+    if (emitterOptions === undefined) {
+      logger.debug(
+        `No emitter options schema found for package ${packageName}, skipping annotation generation`,
+      );
+      return;
+    }
+
+    const configEntries = getConfigEntriesFromEmitterOptions(emitterOptions);
+    if (Object.keys(configEntries).length === 0) {
+      logger.debug(`No configuration entries found for ${packageName}`);
+      return;
+    }
+
+    const packageNodeOptions: { name: string; value: string; comment: string }[] = [];
+    for (const [propertyName, propertyConfig] of Object.entries(configEntries)) {
+      const { value, comment } = propertyConfig as { value: any; comment: string };
+      packageNodeOptions.push({ name: propertyName, value, comment });
+    }
+
+    const parentMap = configYaml.getIn(["options", packageName], true) as any;
+    if (parentMap) {
+      const maxNameLength = Math.max(...packageNodeOptions.map((x) => x.name.length));
+      const maxValueLength = Math.max(...packageNodeOptions.map((x) => String(x.value).length));
+      const commentAlignmentSpacing = 10;
+      parentMap.comment = packageNodeOptions
```

Comment:
i remember you mentioned this comment didn't work as expected. So anything changed here?